fix: add testcast for PutInteger

atgane opened the pull request:
Recreated the PR for [#1277](https://github.com/cilium/ebpf/pull/1277).

ti-mo commented on internal/kconfig/kconfig.go:
```diff
@@ -263,12 +263,25 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
+
 	switch integer.Size {
 	case 1:
+		if integer.Encoding == btf.Signed && (n > math.MaxInt8 && n < math.MaxUint64-math.MaxInt8) {
```
Sorry, but again I don't understand why it needs to be so complicated:
```suggestion
		if integer.Encoding == btf.Signed && (n > math.MaxInt8 || n < math.MinInt8) {
```

atgane replied:
This is because n is of type uint and MinInt8 is of type int, so there was a compiler error in the comparison operation. However, when I cast n to int64, I saw that there was no problem when n exceeded MaxInt64 of int64. I will create a pull request with a similar format by converting n to int64.

ti-mo replied:
> I will create a pull request with a similar format by converting n to int64.

Please, no more pull requests. Just amend and force-push to this PR's branch, please.

atgane replied:
I forced push.

ti-mo commented on internal/kconfig/kconfig.go:
```diff
@@ -263,12 +263,25 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
+		}
 		internal.NativeEndian.PutUint32(data, uint32(n))
 	case 8:
 		internal.NativeEndian.PutUint64(data, uint64(n))
```
This also needs a check, since it's possible for n to overflow/underflow an int64.

atgane replied:
I'll add some test code for the boundary value of an int32.

ti-mo replied:
Never mind this comment, wasn't thinking straight.

ti-mo commented:
@atgane I've squashed your commits and cut down on the size of the test suite. Try to avoid testing error strings, it's of questionable benefit and makes errors annoying to modify.

I split off a few negative tests into TestPutIntegerError.

atgane commented:
> @atgane I've squashed your commits and cut down on the size of the test suite. Try to avoid testing error strings, it's of questionable benefit and makes errors annoying to modify.
> 
> I split off a few negative tests into TestPutIntegerError.

Thank you for your feedback
